src/resolv/async_resolv.c

```diff
@@ -2245,7 +2245,9 @@ static int reply_weight_rearrange(int len,
             new_end = r;
         }
     }
-    new_end->next = NULL;
+    if (new_end) {
+        new_end->next = NULL;
+    }
```

Comment:
Wouldn't it make sense to return an error if new_end is NULL? Or is it OK to just ignore that?